### PR TITLE
feat: Add triangular_solve to jax

### LINKED ISSUE #23374: triangular_solve

### PR DESCRIPTION
**Pull request by omartonsy02**
# PR Description

## Related Issue

Close #23374

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by omartonsy02** [comment]
@jieunboy0516 can you review my pr ?

**Comment by omartonsy02** [comment]
@ivy-leaves can you review my pr?

**Comment by jieunboy0516** [comment]
Hi @omartonsy02 , sorry for the late reply. It seems that your unit tests are failing due to `hypothesis.errors.InvalidArgument: test_jax_triangular_solve() got an unexpected keyword argument 'full_matrices', from `full_matrices=booleans()` in @given` . Would you mind taking a look? Thanks a lot

**Comment by omartonsy02** [comment]
hello, @jieunboy0516 thank you for your review I handled this problem Can you look at it again please

**Comment by jieunboy0516** [comment]
hi @omartonsy02, Thanks for your contribution to Ivy.

- i think that the `triangular_solve` parameters other than `a`  `b` should be taken into consideration as well. 
- the modifiation made in pooling should be removed since this task is only for `triangular_solve`.

Please take a look of the issue when you have time! thx

**Comment by omartonsy02** [comment]
hello @jieunboy0516 thank you for your review  
I think that I am done now
Can you review it again?

**Comment by omartonsy02** [comment]
@jieunboy0516

File: ivy/functional/frontends/jax/lax/linalg.py [hunk 1]
```diff
@@ -38,3 +38,25 @@ def svd(x, /, *, full_matrices=True, compute_uv=True):
...
+    a,
+    b,
+    *,
+    left_side=False,
+    lower=False,
+    transpose_a=False,
+    conjugate_a=False,
+    unit_diagonal=False,
+):
+    return ivy.solve(
```
Comment (jieunboy0516) [inline ivy/functional/frontends/jax/lax/linalg.py:54]: [`ivy.solve`](https://unify.ai/docs/ivy/docs/functional/ivy/linear_algebra/ivy.functional.ivy.linear_algebra.solve.html#solve) does not take these as parameters. Maybe you could take a look of other PRs at the [todo list](https://github.com/unifyai/ivy/issues/1582) such as [this](https://github.com/unifyai/ivy/pull/4585/files) to see how other contributors work on situations like this. Thanks

**Comment by jieunboy0516** [comment]
hi @omartonsy02 , sorry for getting back to you late. I have left a comment. please take a look at the[ official jax doc](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.linalg.triangular_solve.html#jax.lax.linalg.triangular_solve) to see how the function should be implemented. Thanks a lot

**Comment by omartonsy02** [comment]
hello @jieunboy0516  i am sorry for taking your time in my pr can you review it again

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
